projects/npm-tools/packages/npm-scripts/src/utils/getDXPVersion.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+const findRoot = require('./findRoot');
+
+/**
+ * Attempts to infer the current DXP version based on the contents of the
+ * `release.properties` file in the repo root directory.
+ */
+function getDXPVersion() {
+	const modules = findRoot();
+
+	if (modules) {
+		const properties = path.join(
+			path.dirname(modules),
+			'release.properties'
+		);
+
+		try {
+			const contents = fs.readFileSync(properties, 'utf8');
+
+			const [, major, minor, patch] =
+				contents.match(
+					/^\s*release\.info\.version\s*=\s*(\d+)\.(\d+)\.(\d+)\s*$/m
+				) || [];
+
+			if (major !== undefined) {
+				return {
+					major: parseInt(major, 10),
+					minor: parseInt(minor, 10),
+					patch: parseInt(patch, 10),
+				};
+			}
+		}
+		catch {
+
+			// Oh well, it was worth a shot...
```

Comment:
Is it worth adding a log here for debug purposes? I'm sure we won't actually run into it, but just in case.

Reply:
Yeah, my thinking on this one is:

- This should never happen.
- If it does happen, anything we log will get lost in the flood of normal build output.
- If things do go wrong, the "graceful degrade" here is that we just use our latest Babel config, which isn't a bad thing because it means that `master` will behave exactly as it did before. (Presumably, breakage on older branches will be caught by one of our other safety nets, such as manual QA — the thing about breaking IE with arrow functions is that if you fail to transpile them across all of portal, you will notice pretty much immediately as soon as you try anything at all in IE.)